code/modules/events/scrubber_clog.dm

```diff
@@ -0,0 +1,193 @@
+	var/spawn_delay = 10
+	///Used to track/limit produced mobs.
+	var/list/living_mobs = list()
+	///Used for tracking if the clog signal should be sent.
+	var/clogged = TRUE
+
+/datum/round_event/scrubber_clog/announce()
+	priority_announce("Minor biological obstruction detected in the ventilation network. Blockage is believed to be in the [get_area(scrubber)] area.", "Custodial Notification")
+
+/datum/round_event/scrubber_clog/setup()
+	scrubber = get_scrubber()
+	if(!scrubber)
+		CRASH("Unable to find suitable scrubber.")
+
+	RegisterSignal(scrubber, COMSIG_PARENT_QDELETING, .proc/scrubber_move)
+
+	spawned_mob = get_mob()
+	endWhen = rand(300, 600)
+	maximum_spawns = rand(3, 5)
+	spawn_delay = rand(10, 15)
+
+/datum/round_event/scrubber_clog/start() //Sets the scrubber up for unclogging/mob production.
+	scrubber.clog()
+	scrubber.produce_mob() //The first one's free!
+
+/datum/round_event/scrubber_clog/tick() //Checks if spawn_interval is met, then sends signal to scrubber to produce a mob.
+	if(activeFor % spawn_delay == 0 && scrubber.clogged == TRUE)
+		life_check()
+		if(living_mobs.len < maximum_spawns && clogged)
+			scrubber.produce_mob(spawned_mob, living_mobs)
+
+/datum/round_event/scrubber_clog/end() //No end announcement. If you want to take the easy way out and just leave the vent welded, you must open it at your own peril.
+	scrubber.unclog()
+	scrubber = null
+	living_mobs.Cut()
+
+/**
+ * Selects which mob will be spawned for a given scrubber clog event.
+ *
+ * Creates a static list of mobs, which is different based on the severity of the event being run, and returns a pick() of it.
+ */
+
+/datum/round_event/scrubber_clog/proc/get_mob()
+	var/static/list/mob_list = list(
+				/mob/living/simple_animal/mouse,
+				/mob/living/basic/cockroach,
+				/mob/living/simple_animal/butterfly
+				)
+	return(pick(mob_list))
+
+/**
+ * Finds a valid scrubber for the scrubber clog event.
+ *
+ * For every scrubber in the round, checks if the scrubber turf is on-station, and is neither welded nor already clogged, and
+ * adds it to a list. A random scrubber is picked from this list, and returned as the scrubber that will be used for the event.
+ */
+
+/datum/round_event/scrubber_clog/proc/get_scrubber()
+	var/list/scrubber_list = list()
+	for(var/obj/machinery/atmospherics/components/unary/vent_scrubber/scrubber in GLOB.machines)
+		var/turf/scrubber_turf = get_turf(scrubber)
+		if(scrubber_turf && is_station_level(scrubber_turf.z) && !scrubber.welded && !scrubber.clogged)
+			scrubber_list += scrubber
+	return pick(scrubber_list)
+
+/**
+ * Checks which mobs in the mob spawn list are alive.
+ *
+ * Checks each mob in the living_mobs list, to see if they're dead or not. If dead, they're removed from the list.
+ * This is used to keep new mobs spawning as the old ones die.
+ */
+
+/datum/round_event/scrubber_clog/proc/life_check()
+	for(var/mob/living/mob_check in living_mobs)
+		if(mob_check.health <= 0)
+			living_mobs -= mob_check
```

Comment:
because `living_mobs` is full of weakrefs, you will need to resolve them. 
```suggestion
	for(var/datum/weakref/mob_ref as anything in living_mobs)
		var/mob/living/real_mob = mob_ref.resolve()
		if(QDELETED(real_mob) || real_mob.health <= 0) // Note, you can also check for real_mob.stat == DEAD, may be a bit more explicit than health less than zero
			living_mobs -= mob_ref 
```